### PR TITLE
handle scope correctly for @isdefined

### PR DESCRIPTION
**Pull request by vtjnash**
This allows testing whether a variable is defined, and may be especially useful for emitting from a macro (which may want to conditionally initialize a variable).

I expect this will also be very useful when the behavior of dispatch is changed is changed to make static parameters conditionally undefined (https://github.com/JuliaLang/julia/issues/21026#issuecomment-306624369).

### CODE REVIEW

File: src/codegen.cpp [hunk 1]
```diff
@@ -357,6 +357,7 @@ static Function *jlboundserrorv_func;
 static Function *jlcheckassign_func;
 static Function *jldeclareconst_func;
 static Function *jlgetbindingorerror_func;
+static Function *jlboundp_func;
```
Comment (tkelman) [inline src/codegen.cpp]: what is "boundp" supposed to mean?

Comment (vtjnash) [inline src/codegen.cpp]: I don't name the functions, I just use them 😆

Comment (tkelman) [inline src/codegen.cpp]: wow that's old https://github.com/JuliaLang/julia/commit/691b8be811904f6370735f667cd31ed8a4f45d51 and used in a grand total of 3 places before this PR - maybe it should get a new name that says what it does?

Comment (JeffBezanson) [inline src/codegen.cpp]: `boundp` is a common lisp function for testing whether a variable is bound: http://www.ai.mit.edu/projects/iiip/doc/CommonLISP/HyperSpec/Body/fun_boundp.html

Comment (tkelman) [inline src/codegen.cpp]: that explains where the name came from, but doesn't help anyone reading the code who doesn't already know that figure out at a glance what it does

Comment (JeffBezanson) [inline src/codegen.cpp]: The intent of my comment was to explain where the name came from.

Comment (vtjnash) [inline src/codegen.cpp]: What's the conclusion, do I need to add renaming of this function to the PR, or is this gtg?

Comment (tkelman) [inline src/codegen.cpp]: not urgent, we should do a naming-cleanup-pass of the exported API before 1.0 and the unexported APIs after

Comment (JeffBezanson) [inline src/codegen.cpp]: I think the renaming can be done separately.

**Pull request merged**